byron/ledger/executable-spec/src/Ledger/UTxO/Generators.hs

```diff
@@ -0,0 +1,273 @@
+{-# LANGUAGE LambdaCase #-}
+
+module Ledger.UTxO.Generators where
+
+import Control.Applicative (empty)
+import Data.Bitraversable (bitraverse)
+import qualified Data.Map.Strict as M
+
+import Ledger.Core hiding (Range, range)
+import Ledger.UTxO hiding (utxo)
+
+import Hedgehog (Gen, Property, Range, (===), assert, forAll, property)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+import Hedgehog.Internal.Gen
+  (atLeast, ensure, mapGenT, runDiscardEffect, toTree, toTreeMaybeT)
+import Hedgehog.Internal.Tree (NodeT(..), TreeT(..), treeValue)
+import qualified Hedgehog.Internal.Tree as Tree
+
+
+--------------------------------------------------------------------------------
+-- Initial TxOut generator
+--------------------------------------------------------------------------------
+
+-- | Generate a set of initial @TxOut@s from a set of @Addr@s
+genInitialTxOuts :: [Addr] -> Gen [TxOut]
+genInitialTxOuts = Gen.filter (not . null)
+  . genTraverseSubsequence (\a -> TxOut a <$> genLovelace)
+
+genLovelace :: Gen Lovelace
+genLovelace = Lovelace . fromIntegral <$> Gen.word32 (Range.linear 1 10000)
+
+-- | Generate a subsequence of a list of values and traverse the subsequence
```

Comment:
It seems like these belong to a separate package :) Or to `hedgehog` :) I guess for now we can leave them here, but I'm curious about your future plans for these combinators :medal_sports: